### PR TITLE
Add several tests for more code coverage

### PR DESCRIPTION
**Pull request by jverbraeken**
- Relevant Issues: #
- Related Pull Requests: #169 #171 #181 

* * *

## What
This Pull Request adds to the repository the tests that are needed to get maximum code coverage in combination with the tests in a another local branch that I'll push as soon as #181, #169 and #171 are merged

## Why
This Pull Request is needed because we want to know for sure that everything works correctly and tests are a nice tool to verify that

## How
This feature can be viewed/tested within the project by running the tests

## Alternative implementation
Other implementations that I've have considered are

### CODE REVIEW

**Comment by ericcornelissen** [comment]
How do you mean 100% code coverage? Coveralls shows ~70% and from the classes of which the code coverage changed you're still missing [one line](https://coveralls.io/builds/11754214/source?filename=app%2Fsrc%2Fmain%2Fjava%2Fcom%2Fnervousfish%2Fnervousfish%2Fmodules%2Fcryptography%2FKeyGenerationException.java#L44) :wink:

File: app/src/test/java/com/nervousfish/nervousfish/data_objects/SimpleKeyTest.java [hunk 1]
```diff
@@ -98,4 +116,37 @@ public void testHashCodeNotNull() {
...
+            try (ByteArrayInputStream bis = new ByteArrayInputStream(bytes);
+                 ObjectInputStream ois = new ObjectInputStream(bis)) {
+                IKey key1 = (IKey) ois.readObject();
+                assertTrue(key1.getName().equals("foo"));
+                assertTrue(key1.getKey().equals("bar"));
+            } catch (IOException | ClassNotFoundException e) {
+                e.printStackTrace();
+            }
+        } catch (IOException e) {
+            e.printStackTrace();
```
Comment (ericcornelissen) [inline app/src/test/java/com/nervousfish/nervousfish/data_objects/SimpleKeyTest.java]: I think this way the test doesn't fail when the exception is throw? Should you just add a `throws IOException` to the method definition? (Also applies to similar tests in other test classes)

File: app/src/main/java/com/nervousfish/nervousfish/modules/cryptography/KeyGenerationException.java [hunk 2]
```diff
@@ -68,4 +68,6 @@ private Object readResolve() {
             return new KeyGenerationException(this.throwable);
         }
     }
+
+
```
Comment (clenrock) [inline app/src/main/java/com/nervousfish/nervousfish/modules/cryptography/KeyGenerationException.java]: Hmm..

Comment (TheBonheurs) [inline app/src/main/java/com/nervousfish/nervousfish/modules/cryptography/KeyGenerationException.java]: :shipit:

**Pull request merged**